### PR TITLE
Added a reproducer for #10931

### PR DESCRIPTION
**Pull request by rincebrain**
Just wrote a quick reproducer case for the behavior observed in #10931 , integrated into ZTS.

### Motivation and Context

Just wrote a quick reproducer case for the behavior observed in #10931 , integrated into ZTS, because I was surprised nothing had caught this before, and wanted to be sure it would be caught if it's fixed and not happen again.
### Description

Added a test case, verified that the behavior was as described and expected.
### How Has This Been Tested?

Reproduces the problem behavior every time I've run it locally.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [x] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [x]  All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by tpwrules** [comment]
I don't know if the underlying problem is the same, but it appears this tests reading through an mmap a file that was written "traditionally"? The bug I observed was writing through an mmap to a file that is later read "traditionally". Should there be another test for that?

File: tests/zfs-tests/tests/functional/mmap/mmap_read_002_pos.ksh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+log_must readmmap $TESTDIR/test-read-file
+typeset checksum_still=$(sha256digest /$TESTDIR/test-read-file)
+
+[[ "$checksum_still" == "$checksum_new" ]] || \
+    log_fail "checksum stale after rollback"
+[[ "$checksum_still" == "$checksum" ]] || \
+    log_note Meow $(cat /$TESTDIR/test-read-file)
+    log_note "checksum success ($checksum_still == $checksum) ($checksum_new)"
+
+log_must zfs unmount $TESTPOOL/$TESTFS
```
Comment (freqlabs) [inline tests/zfs-tests/tests/functional/mmap/mmap_read_002_pos.ksh:70]: verify_filesys does `zfs umount -a` so this seems redundant here.

**Review by freqlabs — COMMENTED** [review]
The new test script needs to be added to `tests/zfs-tests/tests/functional/mmap/Makefile.am` as well.

File: tests/zfs-tests/tests/functional/mmap/mmap_read_002_pos.ksh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+typeset checksum=$(sha256digest /$TESTDIR/test-read-file)
+log_must zfs snapshot $TESTPOOL/$TESTFS@original
+log_must readmmap $TESTDIR/test-read-file
+log_note "Overwriting file with new contents"
+log_must echo "234561234561234561" > $TESTDIR/test-read-file
+typeset checksum_new=$(sha256digest /$TESTDIR/test-read-file)
+log_must readmmap $TESTDIR/test-read-file
+log_note "Rolling back file with old contents, in theory"
+log_must zfs rollback $TESTPOOL/$TESTFS@original
+log_must readmmap $TESTDIR/test-read-file
```
Comment (freqlabs) [inline tests/zfs-tests/tests/functional/mmap/mmap_read_002_pos.ksh:61]: `readmmap` overwrites the file with random data, so the checksums are not going to match after calling it. This test doesn't make much sense with that in mind.

**Review by freqlabs — CHANGES_REQUESTED** [review]

File: tests/zfs-tests/tests/functional/mmap/mmap_read_002_pos.ksh (hunk 2 above)
Comment (rincebrain) [reply tests/zfs-tests/tests/functional/mmap/mmap_read_002_pos.ksh:61]: You're quite right. More fool me.

Time to throw it out.

**Pull request closed without merging**